swap/swap.go

```diff
@@ -297,14 +302,17 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 		return err
 	}
 
-	// Check if balance with peer crosses the payment threshold
-	// It is the peer with a negative balance who sends a cheque, thus we check
-	// that the balance is *below* the threshold
+	return s.checkPaymentThreshold(swapPeer)
+}
+
+// Check if balance with peer crosses the payment threshold
+// It is the peer with a negative balance who sends a cheque, thus we check
+// that the balance is *below* the threshold
+func (s *Swap) checkPaymentThreshold(swapPeer *Peer) error {
```

Comment:
this function name does not clearly reflect it's role. it says `checkPaymentThreshold`, however it doesn't just `check` the payment threshold but also sends a cheque. also, i would expect this whole function to be holding a lock of some sort since two simultaneous calls are possible and i guess both would emit a cheque, right?

Reply:
I changed the name. This function also assumes that it is called with the peer lock held (in this case the lock is held by `Add` and `handleConfirmMsg`). I added a comment to clarify this.